src/Adapter/MSTest.TestAdapter/Execution/TestExecutionManager.cs

```diff
@@ -320,36 +313,33 @@ private void ExecuteTestsInSource(IEnumerable<TestCase> tests, IRunContext? runC
 
                 for (int i = 0; i < parallelWorkers; i++)
                 {
-                    tasks.Add(_taskFactory(() =>
```

Comment:
I need to look more into this part because it's changing the logic (I am not sure if that's good or not).

Reply:
yeah given i cant compile (and hence cant run locally) i was guessing as to the correct new code.

happy to take direction from you

Reply:
Probably better to keep the old behaviour in here, we are missing tests to be 100% sure of what was the real intent here.